E131 compilation failing on ESP32 - no matching function for call to 'min(int32_t, const int&)'

### The problem

Compiling ESPHome for ESP32-C3 with the E131 component enabled results in a failed compile. The aim is to utilise E131 with Hyperion-NG.

### Which version of ESPHome has the issue?

2023.6.5

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2023.6.3

### What platform are you using?

ESP32

### Board

esp32-c3-devkitm-1 

### Component causing the issue

e131

### Example YAML snippet

```yaml
e131:
  method: unicast

light:
  - platform: neopixelbus
    type: BRG
    variant: WS2811
    pin: GPIO9
    num_leds: 66
    name: "NeoPixel Light"
    effects:
      - e131:
          universe: 1
```


### Anything in the logs that might be useful for us?

```txt
INFO ESPHome 2023.6.5
INFO Reading configuration /config/esphome/dragon-neop1.yaml...
WARNING GPIO9 is a Strapping PIN and should be avoided.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Generating C++ source...
INFO Compiling app...
Processing dragon-neop1 (board: esp32-c3-devkitm-1; framework: arduino; platform: platformio/espressif32@5.3.0)
--------------------------------------------------------------------------------
Library Manager: Installing esphome/AsyncTCP-esphome @ 1.2.2
INFO Installing esphome/AsyncTCP-esphome @ 1.2.2
Unpacking  [####################################]  100%
Library Manager: AsyncTCP-esphome@1.2.2 has been installed!
INFO AsyncTCP-esphome@1.2.2 has been installed!
Library Manager: Installing esphome/ESPAsyncWebServer-esphome @ 2.1.0
INFO Installing esphome/ESPAsyncWebServer-esphome @ 2.1.0
Unpacking  [####################################]  100%
Library Manager: ESPAsyncWebServer-esphome@2.1.0 has been installed!
INFO ESPAsyncWebServer-esphome@2.1.0 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/noise-c @ 0.1.4
INFO Installing esphome/noise-c @ 0.1.4
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.4 has been installed!
INFO noise-c@0.1.4 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10018.1
INFO Installing esphome/libsodium @ 1.10018.1
Unpacking  [####################################]  100%
Library Manager: libsodium@1.10018.1 has been installed!
INFO libsodium@1.10018.1 has been installed!
Library Manager: Installing bblanchon/ArduinoJson @ 6.18.5
INFO Installing bblanchon/ArduinoJson @ 6.18.5
Unpacking  [####################################]  100%
Library Manager: ArduinoJson@6.18.5 has been installed!
INFO ArduinoJson@6.18.5 has been installed!
Library Manager: Installing makuna/NeoPixelBus @ 2.7.3
INFO Installing makuna/NeoPixelBus @ 2.7.3
Unpacking  [####################################]  100%
Library Manager: NeoPixelBus@2.7.3 has been installed!
INFO NeoPixelBus@2.7.3 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
HARDWARE: ESP32C3 160MHz, 320KB RAM, 4MB Flash
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5
Dependency Graph
|-- AsyncTCP-esphome @ 1.2.2
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 2.1.0
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- noise-c @ 0.1.4
|-- ArduinoJson @ 6.18.5
|-- NeoPixelBus @ 2.7.3
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/api_connection.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/api_frame_helper.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/api_pb2.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/api_pb2_service.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/api_server.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/list_entities.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/proto.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/subscribe_state.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/api/user_services.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/captive_portal/captive_portal.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/e131/e131.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/e131/e131_addressable_light_effect.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/e131/e131_packet.cpp.o
Compiling /data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/esp32/core.cpp.o
src/esphome/components/e131/e131_addressable_light_effect.cpp: In member function 'bool esphome::e131::E131AddressableLightEffect::process_(int, const esphome::e131::E131Packet&)':
src/esphome/components/e131/e131_addressable_light_effect.cpp:57:113: error: no matching function for call to 'min(int32_t, const int&)'
       std::min(it->size(), std::min(output_offset + get_lights_per_universe(), output_offset + packet.count - 1));
                                                                                                                 ^
In file included from /data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/char_traits.h:39,
                 from /data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/string:40,
                 from src/esphome/core/component.h:3,
                 from src/esphome/components/e131/e131_addressable_light_effect.h:3,
                 from src/esphome/components/e131/e131_addressable_light_effect.cpp:1:
/data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/stl_algobase.h:195:5: note: candidate: 'template<class _Tp> const _Tp& std::min(const _Tp&, const _Tp&)'
     min(const _Tp& __a, const _Tp& __b)
     ^~~
/data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/stl_algobase.h:195:5: note:   template argument deduction/substitution failed:
src/esphome/components/e131/e131_addressable_light_effect.cpp:57:113: note:   deduced conflicting types for parameter 'const _Tp' ('long int' and 'int')
       std::min(it->size(), std::min(output_offset + get_lights_per_universe(), output_offset + packet.count - 1));
                                                                                                                 ^
In file included from /data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/char_traits.h:39,
                 from /data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/string:40,
                 from src/esphome/core/component.h:3,
                 from src/esphome/components/e131/e131_addressable_light_effect.h:3,
                 from src/esphome/components/e131/e131_addressable_light_effect.cpp:1:
/data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/stl_algobase.h:243:5: note: candidate: 'template<class _Tp, class _Compare> const _Tp& std::min(const _Tp&, const _Tp&, _Compare)'
     min(const _Tp& __a, const _Tp& __b, _Compare __comp)
     ^~~
/data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/stl_algobase.h:243:5: note:   template argument deduction/substitution failed:
src/esphome/components/e131/e131_addressable_light_effect.cpp:57:113: note:   deduced conflicting types for parameter 'const _Tp' ('long int' and 'int')
       std::min(it->size(), std::min(output_offset + get_lights_per_universe(), output_offset + packet.count - 1));
                                                                                                                 ^
In file included from /data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/algorithm:62,
                 from src/esphome/core/optional.h:19,
                 from src/esphome/core/component.h:7,
                 from src/esphome/components/e131/e131_addressable_light_effect.h:3,
                 from src/esphome/components/e131/e131_addressable_light_effect.cpp:1:
/data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/stl_algo.h:3450:5: note: candidate: 'template<class _Tp> _Tp std::min(std::initializer_list<_Tp>)'
     min(initializer_list<_Tp> __l)
     ^~~
/data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/stl_algo.h:3450:5: note:   template argument deduction/substitution failed:
src/esphome/components/e131/e131_addressable_light_effect.cpp:57:113: note:   mismatched types 'std::initializer_list<_Tp>' and 'long int'
       std::min(it->size(), std::min(output_offset + get_lights_per_universe(), output_offset + packet.count - 1));
                                                                                                                 ^
In file included from /data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/algorithm:62,
                 from src/esphome/core/optional.h:19,
                 from src/esphome/core/component.h:7,
                 from src/esphome/components/e131/e131_addressable_light_effect.h:3,
                 from src/esphome/components/e131/e131_addressable_light_effect.cpp:1:
/data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/stl_algo.h:3456:5: note: candidate: 'template<class _Tp, class _Compare> _Tp std::min(std::initializer_list<_Tp>, _Compare)'
     min(initializer_list<_Tp> __l, _Compare __comp)
     ^~~
/data/cache/platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/8.4.0/bits/stl_algo.h:3456:5: note:   template argument deduction/substitution failed:
src/esphome/components/e131/e131_addressable_light_effect.cpp:57:113: note:   mismatched types 'std::initializer_list<_Tp>' and 'long int'
       std::min(it->size(), std::min(output_offset + get_lights_per_universe(), output_offset + packet.count - 1));
                                                                                                                 ^
*** [/data/dragon-neop1/.pioenvs/dragon-neop1/src/esphome/components/e131/e131_addressable_light_effect.cpp.o] Error 1
========================= [FAILED] Took 17.43 seconds =========================
```


### Additional information

_No response_

--- a/esphome/components/e131/e131_addressable_light_effect.cpp
+++ b/esphome/components/e131/e131_addressable_light_effect.cpp
@@ -51,7 +51,7 @@ bool E131AddressableLightEffect::process_(int universe, const E131Packet &packet
   if (universe < first_universe_ || universe > get_last_universe())
     return false;
 
-  int output_offset = (universe - first_universe_) * get_lights_per_universe();
+  int32_t output_offset = (universe - first_universe_) * get_lights_per_universe();
   // limit amount of lights per universe and received
   int output_end =
       std::min(it->size(), std::min(output_offset + get_lights_per_universe(), output_offset + packet.count - 1));

